exec/java-exec/src/main/java/org/apache/drill/exec/planner/sql/handlers/FindLimit0Visitor.java

```diff
@@ -46,6 +51,32 @@ public static boolean containsLimit0(RelNode rel) {
     return visitor.isContains();
   }
 
+  public static DrillRel addLimitOnTopOfLeafNodes(final DrillRel rel) {
+    final RelShuttleImpl shuttle = new RelShuttleImpl() {
+
+      private RelNode addLimitAsParent(RelNode node) {
+        final RexBuilder builder = node.getCluster().getRexBuilder();
+        final RexLiteral offset = builder.makeExactLiteral(BigDecimal.ZERO);
+        final RexLiteral fetch = builder.makeExactLiteral(BigDecimal.ZERO);
+        return new DrillLimitRel(node.getCluster(), node.getTraitSet(), node, offset, fetch);
```

Comment:
I understand in your case, you only putDrillLimitRel. But you may want to make this Visitor more general, such that it could create any kind of LimitRel, including DrillLimitRel, LogicalLimitRel, DrillLimitPrel, etc. You can do that by define a LimitFactory, and pass to this Visitor. This's similar to what other Calcite rule would do, to make the code more general.